circle.yml

```diff
@@ -14,7 +14,7 @@ jobs:
            name: Install build tools
            command: |
              dnf -y install gcc automake autoconf pkgconfig bmake aspell-devel aspell-en libxml2-devel jansson-devel libyaml-devel findutils || :
-             dnf -y install jq
+             dnf -y install jq puppet || :
```

Comment:
I strongly recommend to tie to a minor version of puppet here, rather than getting the latest. Because the latest in fedora 28 repo may change without giving any notice and your tests may fail without any code change in the ctags repo. 

If a developer wants to start supporting a future puppet6 release, she will have to do a code-change in circleci file and also fix all the newly appearing puppet verification errors. Your tests would not use puppet6 before that even though it has been released upstream already. 

For example the code should be 
```
jq puppet-4.10.10-1  || : 
```

See it in action here: 
```
[root@b252d974d2ad /]# yum --showduplicates list puppet
Last metadata expiration check: 0:01:08 ago on Thu Oct 25 22:31:52 2018.
Available Packages
puppet.noarch                                                                                                      4.10.10-1.fc28                                                                                                       fedora
[root@b252d974d2ad /]# yum install puppet-4.10.10-1
Last metadata expiration check: 0:01:18 ago on Thu Oct 25 22:31:52 2018.
No match for argument: puppet-4.10.10-1
Error: Unable to find a match
[root@b252d974d2ad /]# yum install puppet-4.10.10
Last metadata expiration check: 0:01:47 ago on Thu Oct 25 22:31:52 2018.
Dependencies resolved.
==============================================================================================================================================================================================================================================
 Package                                                           Arch                                               Version                                                       Repository                                           Size
==============================================================================================================================================================================================================================================
Installing:
 puppet                                                            noarch                                             4.10.10-1.fc28                                                fedora                                              1.4 M
Installing dependencies:
 augeas-libs                                                       x86_64                                             1.10.1-2.fc28                                                 fedora                                              389 k
 boost-atomic                                                      x86_64                                             1.66.0-8.fc28                                                 updates                                              53 k
 boost-chrono                                                      x86_64                                             1.66.0-8.fc28                                                 updates                                              62 k
 boost-date-time                                                   x86_64                                             1.66.0-8.fc28                                                 updates                                              68 k
 boost-filesystem                                                  x86_64                                             1.66.0-8.fc28                                                 updates                                              88 k
 boost-locale                                                      x86_64                                             1.66.0-8.fc28                                                 updates                                             304 k
 boost-log                                                         x86_64                                             1.66.0-8.fc28                                                 updates                                             486 k
 boost-program-options                                             x86_64                                             1.66.0-8.fc28                                                 updates                                             179 k
 boost-regex                                                       x86_64                                             1.66.0-8.fc28                                                 updates                                             320 k
 boost-system                                                      x86_64                                             1.66.0-8.fc28                                                 updates                                              57 k
 boost-thread                                                      x86_64                                             1.66.0-8.fc28                                                 updates                                              97 k
 cpp-hocon                                                         x86_64                                             0.1.6-7.fc28                                                  fedora                                              419 k
 facter                                                            x86_64                                             3.9.3-4.fc28                                                  fedora                                              623 k
 hiera                                                             noarch                                             3.3.1-3.fc28                                                  fedora                                               34 k
 leatherman                                                        x86_64                                             1.3.0-6.fc28                                                  fedora                                              386 k
 libicu                                                            x86_64                                             60.2-2.fc28                                                   fedora                                              8.8 M
 libselinux-ruby                                                   x86_64                                             2.8-1.fc28                                                    updates                                             135 k
 libselinux-utils                                                  x86_64                                             2.8-1.fc28                                                    updates                                             165 k
 libstdc++                                                         x86_64                                             8.2.1-4.fc28                                                  updates                                             468 k
 ruby                                                              x86_64                                             2.5.1-93.fc28                                                 updates                                              81 k
 ruby-augeas                                                       x86_64                                             0.5.0-16.fc28                                                 fedora                                               30 k
 ruby-facter                                                       x86_64                                             3.9.3-4.fc28                                                  fedora                                               16 k
 ruby-irb                                                          noarch                                             2.5.1-93.fc28                                                 updates                                              98 k
 ruby-libs                                                         x86_64                                             2.5.1-93.fc28                                                 updates                                             2.8 M
 rubygem-json                                                      x86_64                                             2.1.0-105.fc28                                                fedora                                               59 k
 rubygem-openssl                                                   x86_64                                             2.1.0-93.fc28                                                 updates                                             177 k
 rubygem-pathspec                                                  noarch                                             0.2.1-2.fc28                                                  fedora                                               20 k
 rubygem-psych                                                     x86_64                                             3.0.2-93.fc28                                                 updates                                              90 k
 rubygem-rgen                                                      noarch                                             0.7.0-2.fc24                                                  fedora                                               88 k
 rubygem-ruby-shadow                                               x86_64                                             2.5.0-5.fc28                                                  fedora                                               16 k
 rubygems                                                          noarch                                             2.7.6-93.fc28                                                 updates                                             304 k
 rubypick                                                          noarch                                             1.1.1-8.fc28                                                  fedora                                               11 k
 yaml-cpp                                                          x86_64                                             0.6.1-4.fc28                                                  updates                                             169 k
Installing weak dependencies:
 rubygem-bigdecimal                                                x86_64                                             1.3.4-93.fc28                                                 updates                                              91 k
 rubygem-did_you_mean                                              noarch                                             1.2.0-93.fc28                                                 updates                                              86 k
 rubygem-io-console                                                x86_64                                             0.4.6-93.fc28                                                 updates                                              61 k
 rubygem-rdoc                                                      noarch                                             6.0.3-2.fc28                                                  fedora                                              443 k

Transaction Summary
==============================================================================================================================================================================================================================================
Install  38 Packages

Total download size: 19 M
Installed size: 64 M
Is this ok [y/N]: y
Downloading Packages:
(1/38): ruby-augeas-0.5.0-16.fc28.x86_64.rpm                                                                                                                                                                   72 kB/s |  30 kB     00:00
(2/38): hiera-3.3.1-3.fc28.noarch.rpm                                                                                                                                                                          83 kB/s |  34 kB     00:00
(3/38): ruby-facter-3.9.3-4.fc28.x86_64.rpm                                                                                                                                                                   196 kB/s |  16 kB     00:00
(4/38): rubygem-pathspec-0.2.1-2.fc28.noarch.rpm                                                                                                                                                              248 kB/s |  20 kB     00:00
(5/38): rubygem-ruby-shadow-2.5.0-5.fc28.x86_64.rpm                                                                                                                                                           113 kB/s |  16 kB     00:00
(6/38): rubygem-rgen-0.7.0-2.fc24.noarch.rpm                                                                                                                                                                  419 kB/s |  88 kB     00:00
(7/38): rubypick-1.1.1-8.fc28.noarch.rpm                                                                                                                                                                      129 kB/s |  11 kB     00:00
(8/38): augeas-libs-1.10.1-2.fc28.x86_64.rpm                                                                                                                                                                  1.6 MB/s | 389 kB     00:00
(9/38): puppet-4.10.10-1.fc28.noarch.rpm                                                                                                                                                                      1.5 MB/s | 1.4 MB     00:00
(10/38): leatherman-1.3.0-6.fc28.x86_64.rpm                                                                                                                                                                   2.6 MB/s | 386 kB     00:00
(11/38): cpp-hocon-0.1.6-7.fc28.x86_64.rpm                                                                                                                                                                    2.4 MB/s | 419 kB     00:00
(12/38): facter-3.9.3-4.fc28.x86_64.rpm                                                                                                                                                                       1.4 MB/s | 623 kB     00:00
(13/38): libselinux-utils-2.8-1.fc28.x86_64.rpm                                                                                                                                                               360 kB/s | 165 kB     00:00
(14/38): libselinux-ruby-2.8-1.fc28.x86_64.rpm                                                                                                                                                                275 kB/s | 135 kB     00:00
(15/38): ruby-2.5.1-93.fc28.x86_64.rpm                                                                                                                                                                        663 kB/s |  81 kB     00:00
(16/38): rubygem-json-2.1.0-105.fc28.x86_64.rpm                                                                                                                                                               661 kB/s |  59 kB     00:00
(17/38): rubygems-2.7.6-93.fc28.noarch.rpm                                                                                                                                                                    1.4 MB/s | 304 kB     00:00
(18/38): boost-atomic-1.66.0-8.fc28.x86_64.rpm                                                                                                                                                                528 kB/s |  53 kB     00:00
(19/38): boost-chrono-1.66.0-8.fc28.x86_64.rpm                                                                                                                                                                825 kB/s |  62 kB     00:00
(20/38): boost-system-1.66.0-8.fc28.x86_64.rpm                                                                                                                                                                780 kB/s |  57 kB     00:00
(21/38): boost-locale-1.66.0-8.fc28.x86_64.rpm                                                                                                                                                                2.3 MB/s | 304 kB     00:00
(22/38): boost-thread-1.66.0-8.fc28.x86_64.rpm                                                                                                                                                                1.0 MB/s |  97 kB     00:00
(23/38): boost-filesystem-1.66.0-8.fc28.x86_64.rpm                                                                                                                                                            804 kB/s |  88 kB     00:00
(24/38): boost-date-time-1.66.0-8.fc28.x86_64.rpm                                                                                                                                                             858 kB/s |  68 kB     00:00
(25/38): ruby-libs-2.5.1-93.fc28.x86_64.rpm                                                                                                                                                                   2.2 MB/s | 2.8 MB     00:01
(26/38): boost-log-1.66.0-8.fc28.x86_64.rpm                                                                                                                                                                   2.4 MB/s | 486 kB     00:00
(27/38): libstdc++-8.2.1-4.fc28.x86_64.rpm                                                                                                                                                                    3.6 MB/s | 468 kB     00:00
(28/38): boost-regex-1.66.0-8.fc28.x86_64.rpm                                                                                                                                                                 2.2 MB/s | 320 kB     00:00
(29/38): boost-program-options-1.66.0-8.fc28.x86_64.rpm                                                                                                                                                       2.2 MB/s | 179 kB     00:00
(30/38): yaml-cpp-0.6.1-4.fc28.x86_64.rpm                                                                                                                                                                     1.9 MB/s | 169 kB     00:00
(31/38): rubygem-openssl-2.1.0-93.fc28.x86_64.rpm                                                                                                                                                             2.1 MB/s | 177 kB     00:00
(32/38): rubygem-psych-3.0.2-93.fc28.x86_64.rpm                                                                                                                                                               1.2 MB/s |  90 kB     00:00
(33/38): rubygem-bigdecimal-1.3.4-93.fc28.x86_64.rpm                                                                                                                                                          1.2 MB/s |  91 kB     00:00
(34/38): rubygem-did_you_mean-1.2.0-93.fc28.noarch.rpm                                                                                                                                                        1.1 MB/s |  86 kB     00:00
(35/38): rubygem-io-console-0.4.6-93.fc28.x86_64.rpm                                                                                                                                                          828 kB/s |  61 kB     00:00
(36/38): ruby-irb-2.5.1-93.fc28.noarch.rpm                                                                                                                                                                    1.2 MB/s |  98 kB     00:00
(37/38): libicu-60.2-2.fc28.x86_64.rpm                                                                                                                                                                        8.5 MB/s | 8.8 MB     00:01
(38/38): rubygem-rdoc-6.0.3-2.fc28.noarch.rpm                                                                                                                                                                 968 kB/s | 443 kB     00:00
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                                                                                                         4.5 MB/s |  19 MB     00:04
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                                                                                      1/1
  Installing       : libstdc++-8.2.1-4.fc28.x86_64                                                                                                                                                                                       1/38
  Running scriptlet: libstdc++-8.2.1-4.fc28.x86_64                                                                                                                                                                                       1/38
  Installing       : ruby-libs-2.5.1-93.fc28.x86_64                                                                                                                                                                                      2/38
  Installing       : boost-system-1.66.0-8.fc28.x86_64                                                                                                                                                                                   3/38
  Installing       : boost-chrono-1.66.0-8.fc28.x86_64                                                                                                                                                                                   4/38
  Installing       : boost-thread-1.66.0-8.fc28.x86_64                                                                                                                                                                                   5/38
  Installing       : boost-atomic-1.66.0-8.fc28.x86_64                                                                                                                                                                                   6/38
  Installing       : boost-date-time-1.66.0-8.fc28.x86_64                                                                                                                                                                                7/38
  Installing       : boost-filesystem-1.66.0-8.fc28.x86_64                                                                                                                                                                               8/38
  Installing       : ruby-2.5.1-93.fc28.x86_64                                                                                                                                                                                           9/38
  Installing       : rubypick-1.1.1-8.fc28.noarch                                                                                                                                                                                       10/38
  Installing       : rubygem-openssl-2.1.0-93.fc28.x86_64                                                                                                                                                                               11/38
  Installing       : rubygem-psych-3.0.2-93.fc28.x86_64                                                                                                                                                                                 12/38
  Installing       : rubygems-2.7.6-93.fc28.noarch                                                                                                                                                                                      13/38
  Installing       : libicu-60.2-2.fc28.x86_64                                                                                                                                                                                          14/38
  Running scriptlet: libicu-60.2-2.fc28.x86_64                                                                                                                                                                                          14/38
  Installing       : boost-locale-1.66.0-8.fc28.x86_64                                                                                                                                                                                  15/38
  Installing       : boost-regex-1.66.0-8.fc28.x86_64                                                                                                                                                                                   16/38
  Installing       : rubygem-json-2.1.0-105.fc28.x86_64                                                                                                                                                                                 17/38
  Installing       : boost-program-options-1.66.0-8.fc28.x86_64                                                                                                                                                                         18/38
  Installing       : cpp-hocon-0.1.6-7.fc28.x86_64                                                                                                                                                                                      19/38
  Installing       : boost-log-1.66.0-8.fc28.x86_64                                                                                                                                                                                     20/38
  Installing       : leatherman-1.3.0-6.fc28.x86_64                                                                                                                                                                                     21/38
  Running scriptlet: leatherman-1.3.0-6.fc28.x86_64                                                                                                                                                                                     21/38
  Installing       : rubygem-pathspec-0.2.1-2.fc28.noarch                                                                                                                                                                               22/38
  Installing       : rubygem-rgen-0.7.0-2.fc24.noarch                                                                                                                                                                                   23/38
  Installing       : rubygem-ruby-shadow-2.5.0-5.fc28.x86_64                                                                                                                                                                            24/38
  Installing       : rubygem-io-console-0.4.6-93.fc28.x86_64                                                                                                                                                                            25/38
  Installing       : hiera-3.3.1-3.fc28.noarch                                                                                                                                                                                          26/38
  Installing       : ruby-irb-2.5.1-93.fc28.noarch                                                                                                                                                                                      27/38
  Installing       : libselinux-ruby-2.8-1.fc28.x86_64                                                                                                                                                                                  28/38
  Installing       : yaml-cpp-0.6.1-4.fc28.x86_64                                                                                                                                                                                       29/38
  Running scriptlet: yaml-cpp-0.6.1-4.fc28.x86_64                                                                                                                                                                                       29/38
  Installing       : facter-3.9.3-4.fc28.x86_64                                                                                                                                                                                         30/38
  Running scriptlet: facter-3.9.3-4.fc28.x86_64                                                                                                                                                                                         30/38
  Installing       : ruby-facter-3.9.3-4.fc28.x86_64                                                                                                                                                                                    31/38
  Installing       : libselinux-utils-2.8-1.fc28.x86_64                                                                                                                                                                                 32/38
  Installing       : augeas-libs-1.10.1-2.fc28.x86_64                                                                                                                                                                                   33/38
  Running scriptlet: augeas-libs-1.10.1-2.fc28.x86_64                                                                                                                                                                                   33/38
  Installing       : ruby-augeas-0.5.0-16.fc28.x86_64                                                                                                                                                                                   34/38
  Running scriptlet: puppet-4.10.10-1.fc28.noarch                                                                                                                                                                                       35/38
  Installing       : puppet-4.10.10-1.fc28.noarch                                                                                                                                                                                       35/38
  Running scriptlet: puppet-4.10.10-1.fc28.noarch                                                                                                                                                                                       35/38
  Installing       : rubygem-rdoc-6.0.3-2.fc28.noarch                                                                                                                                                                                   36/38
  Installing       : rubygem-bigdecimal-1.3.4-93.fc28.x86_64                                                                                                                                                                            37/38
  Installing       : rubygem-did_you_mean-1.2.0-93.fc28.noarch                                                                                                                                                                          38/38
  Running scriptlet: rubygem-did_you_mean-1.2.0-93.fc28.noarch                                                                                                                                                                          38/38
  Verifying        : puppet-4.10.10-1.fc28.noarch                                                                                                                                                                                        1/38
  Verifying        : hiera-3.3.1-3.fc28.noarch                                                                                                                                                                                           2/38
  Verifying        : ruby-augeas-0.5.0-16.fc28.x86_64                                                                                                                                                                                    3/38
  Verifying        : ruby-facter-3.9.3-4.fc28.x86_64                                                                                                                                                                                     4/38
  Verifying        : rubygem-pathspec-0.2.1-2.fc28.noarch                                                                                                                                                                                5/38
  Verifying        : rubygem-rgen-0.7.0-2.fc24.noarch                                                                                                                                                                                    6/38
  Verifying        : rubygem-ruby-shadow-2.5.0-5.fc28.x86_64                                                                                                                                                                             7/38
  Verifying        : rubypick-1.1.1-8.fc28.noarch                                                                                                                                                                                        8/38
  Verifying        : augeas-libs-1.10.1-2.fc28.x86_64                                                                                                                                                                                    9/38
  Verifying        : facter-3.9.3-4.fc28.x86_64                                                                                                                                                                                         10/38
  Verifying        : cpp-hocon-0.1.6-7.fc28.x86_64                                                                                                                                                                                      11/38
  Verifying        : leatherman-1.3.0-6.fc28.x86_64                                                                                                                                                                                     12/38
  Verifying        : libselinux-utils-2.8-1.fc28.x86_64                                                                                                                                                                                 13/38
  Verifying        : libselinux-ruby-2.8-1.fc28.x86_64                                                                                                                                                                                  14/38
  Verifying        : ruby-libs-2.5.1-93.fc28.x86_64                                                                                                                                                                                     15/38
  Verifying        : ruby-2.5.1-93.fc28.x86_64                                                                                                                                                                                          16/38
  Verifying        : rubygems-2.7.6-93.fc28.noarch                                                                                                                                                                                      17/38
  Verifying        : rubygem-json-2.1.0-105.fc28.x86_64                                                                                                                                                                                 18/38
  Verifying        : boost-atomic-1.66.0-8.fc28.x86_64                                                                                                                                                                                  19/38
  Verifying        : boost-chrono-1.66.0-8.fc28.x86_64                                                                                                                                                                                  20/38
  Verifying        : boost-system-1.66.0-8.fc28.x86_64                                                                                                                                                                                  21/38
  Verifying        : boost-locale-1.66.0-8.fc28.x86_64                                                                                                                                                                                  22/38
  Verifying        : boost-thread-1.66.0-8.fc28.x86_64                                                                                                                                                                                  23/38
  Verifying        : libicu-60.2-2.fc28.x86_64                                                                                                                                                                                          24/38
  Verifying        : boost-filesystem-1.66.0-8.fc28.x86_64                                                                                                                                                                              25/38
  Verifying        : boost-date-time-1.66.0-8.fc28.x86_64                                                                                                                                                                               26/38
  Verifying        : boost-log-1.66.0-8.fc28.x86_64                                                                                                                                                                                     27/38
  Verifying        : boost-regex-1.66.0-8.fc28.x86_64                                                                                                                                                                                   28/38
  Verifying        : libstdc++-8.2.1-4.fc28.x86_64                                                                                                                                                                                      29/38
  Verifying        : boost-program-options-1.66.0-8.fc28.x86_64                                                                                                                                                                         30/38
  Verifying        : yaml-cpp-0.6.1-4.fc28.x86_64                                                                                                                                                                                       31/38
  Verifying        : rubygem-openssl-2.1.0-93.fc28.x86_64                                                                                                                                                                               32/38
  Verifying        : rubygem-psych-3.0.2-93.fc28.x86_64                                                                                                                                                                                 33/38
  Verifying        : rubygem-bigdecimal-1.3.4-93.fc28.x86_64                                                                                                                                                                            34/38
  Verifying        : rubygem-did_you_mean-1.2.0-93.fc28.noarch                                                                                                                                                                          35/38
  Verifying        : rubygem-io-console-0.4.6-93.fc28.x86_64                                                                                                                                                                            36/38
  Verifying        : rubygem-rdoc-6.0.3-2.fc28.noarch                                                                                                                                                                                   37/38
  Verifying        : ruby-irb-2.5.1-93.fc28.noarch                                                                                                                                                                                      38/38

Installed:
  puppet.noarch 4.10.10-1.fc28                rubygem-bigdecimal.x86_64 1.3.4-93.fc28        rubygem-did_you_mean.noarch 1.2.0-93.fc28         rubygem-io-console.x86_64 0.4.6-93.fc28        rubygem-rdoc.noarch 6.0.3-2.fc28
  augeas-libs.x86_64 1.10.1-2.fc28            boost-atomic.x86_64 1.66.0-8.fc28              boost-chrono.x86_64 1.66.0-8.fc28                 boost-date-time.x86_64 1.66.0-8.fc28           boost-filesystem.x86_64 1.66.0-8.fc28
  boost-locale.x86_64 1.66.0-8.fc28           boost-log.x86_64 1.66.0-8.fc28                 boost-program-options.x86_64 1.66.0-8.fc28        boost-regex.x86_64 1.66.0-8.fc28               boost-system.x86_64 1.66.0-8.fc28
  boost-thread.x86_64 1.66.0-8.fc28           cpp-hocon.x86_64 0.1.6-7.fc28                  facter.x86_64 3.9.3-4.fc28                        hiera.noarch 3.3.1-3.fc28                      leatherman.x86_64 1.3.0-6.fc28
  libicu.x86_64 60.2-2.fc28                   libselinux-ruby.x86_64 2.8-1.fc28              libselinux-utils.x86_64 2.8-1.fc28                libstdc++.x86_64 8.2.1-4.fc28                  ruby.x86_64 2.5.1-93.fc28
  ruby-augeas.x86_64 0.5.0-16.fc28            ruby-facter.x86_64 3.9.3-4.fc28                ruby-irb.noarch 2.5.1-93.fc28                     ruby-libs.x86_64 2.5.1-93.fc28                 rubygem-json.x86_64 2.1.0-105.fc28
  rubygem-openssl.x86_64 2.1.0-93.fc28        rubygem-pathspec.noarch 0.2.1-2.fc28           rubygem-psych.x86_64 3.0.2-93.fc28                rubygem-rgen.noarch 0.7.0-2.fc24               rubygem-ruby-shadow.x86_64 2.5.0-5.fc28
  rubygems.noarch 2.7.6-93.fc28               rubypick.noarch 1.1.1-8.fc28                   yaml-cpp.x86_64 0.6.1-4.fc28

Complete!
[root@b252d974d2ad /]#
```